File: Loading Views/LoadingViews/Presentations Option 0 Pop Over/PopOverViewController.m
//
//  PopOverViewController.m
//  LoadingViews
//
//  Created by Luis Castillo on 12/30/15.
//  Copyright © 2015 DevBros. All rights reserved.
//

#import "PopOverViewController.h"

@interface PopOverViewController ()

@end

@implementation PopOverViewController

@synthesize popUpView, logoImage, messageLabel;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    
    if (self) {
            // Custom initialization
    }
    return self;
}

    //MARK: View Loading
- (void)viewDidLoad {
    
    [self debug];

//    CGRect mainscreen_bounds = [UIScreen mainScreen] .bounds;
//    CGFloat navBarHeight = super.navigationController.navigationBar.frame.size.height;
//    if (navBarHeight > 0)
//    {
//        CGFloat mainscreen_UpdatedHeight = mainscreen_bounds.size.height-navBarHeight-1;
//        
//        
//        self.view.frame = mainscreen_bounds;
//    }
//    else
//    {
//        self.view.frame = mainscreen_bounds;
//    }
    
    
    
    [super viewDidLoad];

    
    self.view.backgroundColor           = [[UIColor blackColor] colorWithAlphaComponent:.6];
    self.popUpView.layer.cornerRadius   = 5;
    self.popUpView.layer.shadowOpacity  = 0.8;
    self.popUpView.layer.shadowOffset   = CGSizeMake(0.0f, 0.0f);

    [self debug];
}//eom

//MARK: Animations
-(void) showAnimation
{
    self.view.transform = CGAffineTransformMakeScale(1.3, 1.3);
    self.view.alpha = 0;
    [UIView animateWithDuration:.25 animations:^{
        self.view.alpha = 1;
        self.view.transform = CGAffineTransformMakeScale(1, 1);
    }];
    
}//eom


-(void) removeAnimation
{
    [UIView animateWithDuration:.25 animations:^{
        self.view.transform = CGAffineTransformMakeScale(1.3, 1.3);
        self.view.alpha = 0.0;
    } completion:^(BOOL finished) {
        if (finished) {
            [self.view removeFromSuperview];
        }
    }];
}//eom

- (IBAction)closePopup:(id)sender {
    [self removeAnimation];
}//eo-a



- (void)showInView:(UIView *)aView
        withImage:(UIImage*) imageProvided
       withMessage:(NSString *)msgProvided
          animated:(BOOL)animated
{
    dispatch_async(dispatch_get_main_queue(), ^{
        [aView addSubview:self.view];
        self.logoImage.image = imageProvided;
        self.messageLabel.text = msgProvided;
        if (animated) {
            [self showAnimation];
        }
    });
}


-(void)debug
{
    CGRect bounds_mainscreen = [UIScreen mainScreen] .bounds;
    CGRect vc_bounds = self.view.bounds;
    CGRect vc_frame = self.view.frame;
    
    NSLog(@"[main screen - bounds] x,y (%f, %f)",bounds_mainscreen.origin.x, bounds_mainscreen.origin.y);
    NSLog(@"[main screen - bounds] width,height (%f,%f)",bounds_mainscreen.size.width, bounds_mainscreen.size.height );
    
    NSLog(@"[VC - bounds] x,y (%f, %f)",vc_bounds.origin.x, vc_bounds.origin.y);
    NSLog(@"[VC - bounds] width,height (%f,%f)",vc_bounds.size.width, vc_bounds.size.height );
    
    NSLog(@"[VC - frame] x,y (%f, %f)",vc_frame.origin.x, vc_frame.origin.y);
    NSLog(@"[VC - frame] width,height (%f,%f)",vc_frame.size.width, vc_frame.size.height );
    
}//eom
@end

File: Loading Views/LoadingViews/Presentations Option 0 Pop Over/PopOverViewController.swift
//
//  PopOverViewController.swift
//  LoadingViews
//
//  Created by Luis Castillo on 12/30/15.
//  Copyright © 2015 devbros. All rights reserved.
//

import UIKit
import QuartzCore


class PopOverViewControllerSwift: UIViewController {

    @IBOutlet weak var viewToDisplay: UIView!
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: Bundle!) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }
    
    
    //MARK: View Loading
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.black.withAlphaComponent(0.6)
        self.viewToDisplay.layer.cornerRadius = 5
        self.viewToDisplay.layer.shadowOpacity = 0.8
        self.viewToDisplay.layer.shadowOffset = CGSize(width: 0.0, height: 0.0)
        
        
        self.setupGesturesToCloseView()
        
    }
    
    //MARK: Show View
    
    func showInView(_ aView: UIView!, animated: Bool)
    {
        aView.addSubview(self.view)
        if animated
        {
            self.showAnimate()
        }
    }
    

    //MARK: Animations
    
    func showAnimate()
    {
        self.view.transform = CGAffineTransform(scaleX: 1.3, y: 1.3)
        self.view.alpha = 0.0;
        UIView.animate(withDuration: 0.25, animations: {
            self.view.alpha = 1.0
            self.view.transform = CGAffineTransform(scaleX: 1.0, y: 1.0)
        });
    }
    
    func removeAnimate()
    {
        UIView.animate(withDuration: 0.25, animations: {
            self.view.transform = CGAffineTransform(scaleX: 1.3, y: 1.3)
            self.view.alpha = 0.0;
            }, completion:{(finished : Bool)  in
                if (finished)
                {
                    self.view.removeFromSuperview()
                }
        });
    }
    
    //MARK: Close
    func closeView()
    {
        self.removeAnimate()
    }//eom
    
    //MARK: Gesture Setup
    func setupGesturesToCloseView()
    {
//        //double tap
//        let tap = UITapGestureRecognizer.init(target: self, action: "closeView")
//        self.view .addGestureRecognizer(tap)
        
        //swipes
        let swipe = UISwipeGestureRecognizer.init(target: self,
                                                  action: #selector(PopOverViewControllerSwift.closeView as (PopOverViewControllerSwift) -> () -> ()))
        swipe.direction = UISwipeGestureRecognizerDirection.right
        
        self.view .addGestureRecognizer(swipe)
        
    }//eom
    
    @IBAction func closeView(_ sender: AnyObject)
    {
        self.closeView()
    }//eo-a
    
    
    //MARK: Memory
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - Navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }

}
